lib/ExternalModuleFactoryPlugin.js

```diff
@@ -27,10 +27,23 @@ class ExternalModuleFactoryPlugin {
 					}
 					if (value === false) return factory(data, callback);
 					if (value === true) value = dependency.request;
-					if (type === undefined && /^[a-z0-9]+ /.test(value)) {
-						const idx = value.indexOf(" ");
-						type = value.substr(0, idx);
-						value = value.substr(idx + 1);
+					if (type === undefined) {
+						if (typeof value === "string" && /^[a-z0-9]+ /.test(value)) {
+							const idx = value.indexOf(" ");
+							type = value.substr(0, idx);
+							value = value.substr(idx + 1);
+						} else if (typeof value === "object" && !Array.isArray(value)) {
+							const keys = Object.keys(value);
+							if (keys.length === 1) {
+								type = keys[0];
+								value = value[type];
+							}
+						}
+					}
+					if (type === "amd" && !globalType.startsWith("amd")) {
+						throw new Error(
+							"AMD dependencies are only supported for AMD targets"
```

Comment:
Add a test for the error case

Reply:
Could you point me at an existing error case that's relatively similar that I could follow?